src/common/hooks/useVerifyAppToken.tsx

```diff
@@ -0,0 +1,46 @@
+import { useEffect, useState } from "react";
+import { KJUR } from "jsrsasign";
+import forge from "node-forge";
+
+const publicKeyUrl = import.meta.env.VITE_PUBLIC_KEY_BASE_URL || "https://app.contentstack.com";
+
+export const useVerifyAppToken = (token: string | null) => {
+  const [isValidAppToken, setIsValidAppToken] = useState<boolean | null>(null);
+
+  useEffect(() => {
+    const verify = async () => {
+      if (!token) {
+        setIsValidAppToken(true); // Skip verification if no token is present
```

Comment:
why are we doing this? First we don't need to setState it's unnecessary . we can direct return true outside of useEffect